### ISSUE TITLE
Review custom x-ob-account-id header passed to RS for create payment flows

### ISSUE THREAD

**Issue by dbadham-fr** [SecureBankingAccessToolkit/SecureBankingAccessToolkit#542]
`x-ob-account-id` is a custom header that we have added to the API controllers for payments APIs.

Review the purpose of this header, as we are customising the OB API which may be confusing for clients.

**Comment by dbadham-fr** [comment]
`x-ob-account-id` this is passed for the following APIs:

- DomesticStandingOrder
- DomesticScheduledPayments
- InternationalStandingOrder
- InternationalScheduledPayment

Interestingly, it is not passed for the following:

- DomesticPayments
- InternationalPayments
- FilePayments
- DomesticVrp

One would assume that if we needed this AccountId to make a payment then it would be present for all flows.

**Comment by frlaura-jianu** [comment]
In the API Controllers for Domestic Scheduled Payments, International Scheduled Payments, Domestic Standing Orders, International Standing Orders the payment data is stored for the Accounts API versus the other APIs where there is no need to store the payment data.

There are designed services in RS for both types of APIs (domestic/international scheduled payments & domestic/international standing orders) to save, retrieve and update both scheduled payments & standing orders for the Accounts API.

In the Accounts API both standing orders & scheduled payments resources can be retrieved for a specific account identified by `AccountId` or for all accounts that the PSU has consented to.
https://openbankinguk.github.io/read-write-api-site3/v3.1.8/resources-and-data-models/aisp/standing-orders.html#overview
https://openbankinguk.github.io/read-write-api-site3/v3.1.8/resources-and-data-models/aisp/scheduled-payments.html#overview

The value of the `x-ob-account-id` header is set to that of the `AccountId` field in the IDM object for the payment intent, used in the services earlier mentioned on the creation & storing of a persisted scheduled payment/standing order within the repository.

In conclusion, the `x-ob-account-id` header has to be kept.
See also #565.
